EpikCord/client/websocket.py

```diff
@@ -70,21 +70,22 @@ def __init__(self, client: WebSocketClient):
             OpCode.HEARTBEAT_ACK: self.heartbeat_ack,
         }
 
-    def event(self):
+    def event(self, func: AsyncFunction):
         """Register an event handler. This is a decorator."""
 
-        def decorator(func: AsyncFunction):
-            name = func.__name__.lower().replace("on_", "")
-            self.events[name].append(func)
-            return func
-
-        return decorator
+        name = func.__name__.lower().replace("on_", "")
```

Comment:
no a fan on forcing user function naming.

Would prefer a approach such as the following, that allows the user to give a proper name.

```python
# client bot

@bot.event(EventType.ON_READY)
def initialize_channel_update(client):
    pass
```

Reply:
I don't want to have an enum dedicated to each and every single event name, and update that as time goes on.

Reply:
no sure to see the problem

Reply:
Sure, there is an (1) issue with my implementation, but I see two in yours. I would have to create an Enum of EVENT_NAME_IN_SCREAMING_SNAKE_CASE to their also SCREAMING_SNAKE_CASE or even just snake_case variant as a string, and that makes the event decorator required to be called, an issue in the current version of the codebase I want to tackle.

Reply:
I'm not sure why you would need a "variant" in snake_case, and making the decorator mandatory seems like a good thing tbh

Reply:
```py
from enum import Enum

class EventName(Enum):
    MESSAGE_CREATE = auto()
    @property
    def event_name(self):
        return self.name.lower()

print(EventName.MESSAGE_CREATE.event_name) # => message_create
```

Reply:
I still would have to create an entire list of event names. If there are more pros than cons I might do this.